src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/coding/unnecessaryparentheses/InputUnnecessaryParenthesesCheckSwitchExpression.java

```diff
@@ -18,22 +18,22 @@
     MathOperation2 tooManyParens(int k) {
         return switch (k) {
             case 1 -> {
-                MathOperation2 case5 = (a, b) -> (a + b); // violation
+                MathOperation2 case5 = (a, b) -> (a + b); // violation '.* right-hand side'
```

Comment:
The rationale behind specifying violation messages in input files was to make them easier to read and avoid traveling back to the test file to see what the actual violation is, `.* right-hand side` doesn't clearly imply what the violation message could be. Here the original message is-
`Unnecessary parentheses around assignment right-hand side.`
You can shorten it to-
`Unnecessary paren.* around assignment right.* side` or to anything else which implies what the actual violation is.

If the line length doesn't cooperate then you could shorten the variables name, wrap it on the next line or use violation above or below pattern (last resort)

Reply:
violation above/below pattern should be considered as the very last resort. The case which you have mentioned is a chained method call-
```java
new HashSet<Integer>().stream().filter((filter) -> filter > 0); // violation '.* lambda .*'
```
It is good to wrap chained method calls for readability, I would have done something like-
```java
new HashSet<Integer>()
    .stream()
    .filter((filter) -> filter > 0);  // violation .........
```
or at least would have wrapped `filter` onto the next line (as some prefer to keep `stream()` on the first line).
> will introduce a new line and I will have to make changes in the expected line numbers in the test, right? Can I do that?

Absolutely, you can modify the test file.

Reply:
Change indentation from +4 to +2, it will give extra space for better message

Reply:
Rename MathOperation2 to MathOp2 to give you even more space.
Anything that is not about parentheses are ok to change/shorten

Reply:
Done

Reply:
Resolved